src/credentials/providers/credentialsProviderManager.ts

```diff
@@ -26,6 +26,19 @@ export class CredentialsProviderManager {
         return providers
     }
 
+    /**
+     * Returns a map of profile names to credential ids from all credential
+     * sources.
+     */
+    public async getProfiles() {
```

Comment:
* declare the return type in the signature
* profile is the wrong term. profile is shared credentials specific, the generalized term is credentials. `getAllCredentials` is a more suitable name.

Reply:
it's a map of profile names to credentials. But no objection.